buildSrc/plugins/src/main/kotlin/org/jetbrains/kotlin/KonanTestSuite.kt

```diff
@@ -0,0 +1,139 @@
+package org.jetbrains.kotlin
+
+import java.io.PrintWriter
+import java.io.StringWriter
+
+enum class TestStatus {
+    PASSED,
+    FAILED,
+    ERROR,
+    SKIPPED
+}
+
+data class TestResult(val status: TestStatus, val comment: String = "")
+
+data class Statistics(
+        var passed: Int = 0,
+        var failed: Int = 0,
+        var error: Int = 0,
+        var skipped: Int = 0) {
+
+    val total: Int
+        get() = passed + failed + error + skipped
+
+    fun pass(count: Int = 1) { passed += count }
+
+    fun skip(count: Int = 1) { skipped += count }
+
+    fun fail(count: Int = 1) { failed += count }
+
+    fun error(count: Int = 1) { error += count }
+
+    fun add(other: Statistics) {
+        passed += other.passed
+        failed += other.failed
+        error += other.error
+        skipped += other.skipped
+    }
+}
+
+open class KonanTestSuite(val name: String, val statistics: Statistics) {
```

Comment:
abstract?

Reply:
No, there `KonanTestSuite` used to be default implementation for command line launches and so on. The single child is for TeamCity execution.